chain/jsonrpc/src/lib.rs

```diff
@@ -131,25 +134,32 @@ impl JsonRpcHandler {
 
     fn send_tx_commit(&self, params: Option<Value>) -> Box<Future<Item = Value, Error = RpcError>> {
         let tx = ok_or_rpc_error!(parse_tx(params));
-        let _hash = tx.get_hash();
+        let tx_hash = tx.get_hash();
+        let view_client_addr = self.view_client_addr.clone();
         Box::new(
             self.client_addr
                 .send(NetworkClientMessages::Transaction(tx))
-                .and_then(|_result| {
-                    // TODO: implement waiting for transaction hash to appear in client or some timeout without blocking.
-                    unimplemented!();
-                    Ok(Value::Null)
-                    //                let check_result = Interval::new(Instant::now(), Duration::from_secs(1)).take(5).for_each(|()| {
-                    //                    self.client_addr.send()
-                    //                }).wait();
-                    //                match check_result {
-                    //                    Ok(result) => {
-                    //                        Ok(Value::Null)
-                    //                    },
-                    //                    Err(result) => Err(RpcError::server_error(Some(format!("Waiting for {} timed out.", hash))))
-                    //                }
+                .map_err(|err| RpcError::server_error(Some(err.to_string())))
+                .and_then(move |_result| {
+                    Interval::new_interval(std::time::Duration::from_millis(100))
+                        .map(move |_| {
+                            view_client_addr.send(TxStatus { tx_hash }).wait()
+                        })
+                        .filter(|tx_status| {
+                            match tx_status {
+                                Ok(Ok(tx_status)) => {
+                                    match tx_status.status {
+                                        FinalTransactionStatus::Unknown => false,
```

Comment:
FinalTransactionStatus can be Started (e.g. while first tx already in the block, but there are subsequent recepts), so it should be consdered as `false` as well.

Reply:
@ilblackdragon Done